drivers/i3c/i3c_common.c

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/slist.h>
+
+#include <zephyr/drivers/i3c.h>
+
+#define LOG_LEVEL CONFIG_I3C_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(i3c);
+
+void i3c_dump_msgs(const char *name, const struct i3c_msg *msgs,
+		   uint8_t num_msgs, struct i3c_device_desc *target)
+{
+	LOG_DBG("I3C msg: %s, addr=%x", name, target->dynamic_addr);
+	for (unsigned int i = 0; i < num_msgs; i++) {
+		const struct i3c_msg *msg = &msgs[i];
+
+		LOG_DBG("   %c len=%02x: ",
+			msg->flags & I3C_MSG_READ ? 'R' : 'W', msg->len);
+		if (!(msg->flags & I3C_MSG_READ)) {
+			LOG_HEXDUMP_DBG(msg->buf, msg->len, "contents:");
+		}
+	}
+}
+
+void i3c_addr_slots_status_set(struct i3c_addr_slots *slots,
+			       uint8_t dev_addr,
+			       enum i3c_addr_slot_status status)
+{
+	int bitpos;
+	int idx;
+
+	__ASSERT_NO_MSG(slots != NULL);
```

Comment:
note here, afaik we tipically not assert for programming errors like this one.

Reply:
This is a new API, so I thought it could help the early adopters. :)